spec/controllers/tasks_controller_spec.rb

```diff
@@ -1041,4 +1041,76 @@
       end.to change(collection.tasks, :count).by(+1)
     end
   end
+
+  describe 'POST #generate_test' do
+    let(:task_user) { create(:user) }
+    let(:access_level) { :public }
+    let(:task) { create(:task, user: task_user, access_level:) }
+
+    before do
+      sign_in task_user
+      Settings.open_ai.access_token = 'access_token'
+    end
+
+    context 'when GptGenerateTests is successful' do
+      before do
+        allow(TaskService::GptGenerateTests).to receive(:call)
+        post :generate_test, params: {id: task.id}
+      end
+
+      after do
+        Settings.open_ai.access_token = nil
+      end
```

Comment:
Rather than repeating this `after` block multiple times, wouldn't it work to move it below setting the access token and remove the duplications?

```rb
before do
   sign_in task_user
   Settings.open_ai.access_token = 'access_token'
 end
```